lib/state/revisions/reducer.js

```diff
@@ -0,0 +1,16 @@
+import { combineReducers } from 'redux';
+
+const revisionState = ( state = { revision: null, isViewingRevisions: false }, action ) => {
+	if ( 'setRevisionState' !== action.type ) {
+		return state;
+	}
+	console.log( action );
+	return {
+		...state,
+		...action.revisionState,
+	};
+};
+
+export default combineReducers( {
+	revisionState,
+} );
```

Comment:
the main thing here is to recognize what is intrinsically coupled and what isn't. the way this is written implies that the two state properties are intrinsically bound, but I don't think they are. we can do a simple extraction to split these reducers into simpler and much more _testable_ things

```js
export const revision = ( state = null, { type, revisionState } ) =>
	REVISION_STATE_SET === type
		? revisionState
		: state;

export const isViewingRevisions = ( state = false, { type, revisionState } ) =>
	REVISION_STATE_SET === type
		? revisionState === null
		: state;

export default combineReducers( {
	isViewingRevision,
	revision,
} );
```

Now we have two separate reducers each with simple data types for state and they are so stupid simple that we practically don't even need to test them properly in order to benefit form their ability to be easily tested.